_includes/langselect.html

```diff
@@ -1,17 +1,218 @@
-<div id="langselect">
-  <div lang="en" id="suglang">This site might also be available in your native language. Select your preferred language below:</div>
-  <ul>
-    {% for lng in site.languages %}
-      {% if lng == site.active_lang %}
-        <li class="selected" data-slang="{{ lng }}">{{ lng }}</li>
-      {% else %}
-        {% if lng == site.default_lang %}
-          {% capture langlink %} {{ page.url }}{% endcapture %}
-        {% else %}
-          {% capture langlink %}/{{ lng }}{{ page.url }}{% endcapture %}
-        {% endif %}
-        <li><a href="{{ langlink }}" data-slang="{{ lng }}">{{ lng }}</a></li>
-      {% endif %}
-    {% endfor %}
-  </ul>
+<div id="langselect-container">
+  <div id="langselect">
+    <!-- Language Dropdown -->
+    <div class="language-dropdown">
+      <button class="dropdown-button">
+        <span class="dropdown-button-content">
+          <img src="/assets/img/lang-icon.png" alt="Language Icon" class="lang-icon">
+          <span class="lang-text">{{ site.language_names[site.active_lang] }} &#9660;</span>
+        </span>
+      </button>
+      <div class="dropdown-content">
+        <ul>
+          {% assign sorted_languages = site.languages | sort %}
+          {% for lng in sorted_languages %}
+            {% if lng == site.active_lang %}
+              <li class="selected" data-slang="{{ lng }}">{{ site.language_names[lng] }}</li>
+            {% else %}
+              {% if lng == site.default_lang %}
+                {% capture langlink %}{{ page.url }}{% endcapture %}
+                {% capture lang_rel %}canonical{% endcapture %}
+              {% else %}
+                {% capture langlink %}/{{ lng }}{{ page.url }}{% endcapture %}
+                {% capture lang_rel %}alternate{% endcapture %}
+              {% endif %}
+              <li><a href="{{ langlink }}" rel="{{ lang_rel }}" data-slang="{{ lng }}">{{ site.language_names[lng] }}</a></li>
+            {% endif %}
+          {% endfor %}
+        </ul>
+      </div>
+    </div>
+  </div>
 </div>
+<script>
+  document.addEventListener('DOMContentLoaded', function() {
```

Comment:
Initially I didn't implement something like this on purpose: users may want to switch with a more conscious choice and google should redirect to the correct language by default if the meta tags are set correctly.

Anyway: The code looks a bit too large and complex for a feature like that. I'll have a look at this some time.